processmanager/manager.go

```diff
@@ -237,54 +237,49 @@ func (pm *ProcessManager) ConvertTrace(trace *host.Trace) (newTrace *libpf.Trace
 			log.Errorf("Unexpected frame type 0x%02X (neither error nor interpreter frame)",
 				uint8(frame.Type))
 		case libpf.Native, libpf.Kernel:
-			// When unwinding stacks, the address is obtained from the stack
-			// which contains pointer to the *next* instruction to be executed.
+			// The BPF code classifies whether an address is a return address or not.
+			// Return addresses are where execution resumes when returning to the stack
+			// frame and point to the **next instruction** after the call instruction
+			// that produced the frame.
 			//
-			// For all kernel frames, the kernel unwinder will always produce
-			// a frame in which the RIP is after a call instruction (it hides the top
-			// frames that leads to the unwinder itself).
+			// For these return addresses we subtract 1 from the address in order to
+			// make it point into the call that precedes it: the instruction at the
+			// return address may already be part of whatever code follows after the
+			// call, and we want the return addresses to resolve to the call itself
+			// during symbolization.
 			//
-			// For leaf user mode frames (without kernel frames) the RIP from
-			// our unwinder is good as is, and must not be altered because the
-			// previous instruction address is unknown -- we might have just
-			// executed a jump or a call that got us to the address found in
-			// these frames.
-			//
-			// For other user mode frames we are at the next instruction after a
-			// call. And often the next instruction is already part of the next
-			// source code line's debug info areas. So we need to fixup the non-top
-			// frames so that we get source code lines pointing to the call instruction.
-			// We would ideally wish to subtract the size of the instruction from
-			// the return address we retrieved - but the size of calls can vary
-			// (indirect calls etc.). If, on the other hand, we subtract 1 from
-			// the address, we ensure that we fall into the range of addresses
-			// associated with that function call in the debug information.
-			//
-			// The unwinder will produce stack traces like the following:
-			//
-			// Frame 0:
-			// bla %reg           <- address of frame 0
-			// retq
-			//
-			// Frame 1:
-			// call <function>
-			// add %rax, %rbx     <- address of frame 1 == return address of frame 0
-
+			// Optimally we'd subtract the size of the call instruction here instead
+			// of doing `- 1`, but disassembling backwards is quite difficult for
+			// variable length instruction sets like X86.
 			relativeRIP := frame.Lineno
 			if frame.ReturnAddress {
 				relativeRIP--
 			}
+
+			// Locate mapping info for the frame.
+			var mappingStart, mappingEnd libpf.Address
+			var fileOffset uint64
+			if frame.Type.Interpreter() != libpf.Kernel {
```

Comment:
Should we extend this exception?
E.g. for python, `fileID` is "artificial" and there is no real backing memory mapping:
https://github.com/elastic/otel-profiling-agent/blob/9a7f99b6b67eae165a2b67b314262abfec6227cc/interpreter/python/python.go#L570-L578

or Perl:
https://github.com/elastic/otel-profiling-agent/blob/9a7f99b6b67eae165a2b67b314262abfec6227cc/interpreter/perl/instance.go#L411-L419

Reply:
This code lives in the `case libpf.Native, libpf.Kernel:` branch, so these are filtered out already.

Reply:
ah right - we are in this big switch :+1: